crates/sui-json-rpc-types/src/sui_transaction.rs

```diff
@@ -180,6 +180,10 @@ impl SuiTransactionBlockResponseOptions {
         }
     }
 
+    #[deprecated(
```

Comment:
There is more we could do with the flags here (the criteria for requiring effects etc seems too generous), but I am resisting here seeing as we will be moving to a new SDK soon.

Reply:
Does `default_execution_request_type` need to be updated?

Reply:
I consider it but decided not to because I'm afraid it would open a can of worms that we would then need to deal with in the near term, even though the underlying issues would disappear with the GraphQL Rust SDK, namely:

- Why do we set `WaitForLocalExecution` on the basis of `require_effects` instead of ...`require_local_execution`?
- Why do we need to do anything special to get the raw effects?

It seems at some point these two things got conflated, and now it's easier to keep them coupled until we introduce GraphQL Rust SDK (or we get rid of `WaitForLocalExecution` as a flag on the client), because otherwise we might break the experience for people who are implicitly relying on read-write consistency they got from requesting just effects.